tests/e2e/scenarios/upgrade/run-test

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "CLOUD_PROVIDER=${CLOUD_PROVIDER}"
+
+REPO_ROOT=$(git rev-parse --show-toplevel);
+
+KUBETEST2_COMMON_ARGS="-v=2 --cloud-provider=${CLOUD_PROVIDER} --cluster-name=${CLUSTER_NAME:-} --kops-binary-path=${REPO_ROOT}/bazel-bin/cmd/kops/linux-amd64/kops"
+KUBETEST2_COMMON_ARGS="${KUBETEST2_COMMON_ARGS} --admin-access=${ADMIN_ACCESS:-}"
+
+export GO111MODULE=on
+go get sigs.k8s.io/kubetest2@latest
+go get sigs.k8s.io/kubetest2/kubetest2-tester-ginkgo@latest
+
+cd ${REPO_ROOT}/tests/e2e
+go install ./kubetest2-kops
+
+kubetest2 kops ${KUBETEST2_COMMON_ARGS} --build --kops-root=${REPO_ROOT} --stage-location=${STAGE_LOCATION:-}
+
+RUN_KUBETEST2_DOWN=y
+function kubetest2_down {
+  kubetest2 kops ${KUBETEST2_COMMON_ARGS} --down
+}
+function finish {
+  if [[ "${RUN_KUBETEST2_DOWN}" == "y" ]]; then
+    kubetest2_down || echo "kubetest2 down failed"
+  fi
+}
+trap finish EXIT
+
+kubetest2 kops ${KUBETEST2_COMMON_ARGS} --up
+
+KUBECONFIG=${HOME}/.kube/config
+TEST_ARGS="--kubeconfig=${KUBECONFIG}"
+if [[ "${CLOUD_PROVIDER}" == "aws" ]]; then
+  ZONES=`kops get cluster ${CLUSTER_NAME} -ojson | jq -r .spec.subnets[].zone`
+  CLUSTER_TAG="${CLUSTER_NAME}"
+  TEST_ARGS="${TEST_ARGS} --provider=aws --gce-zone=${ZONES[0]} --cluster-tag=${CLUSTER_TAG}"
```

Comment:
Might be worthy of a comment that --gce-zone is actually provider agnostic (this tripped me up for a minute)

https://github.com/kubernetes/test-infra/blob/1a5044709fb8b5350c99a2a8e5ac208827a04c07/kubetest/kops.go#L692-L697

Reply:
Good call - added